### ISSUE TITLE
⚡ perf(bench): add training throughput benchmark (tokens/sec)

### ISSUE THREAD

**Issue by nishide-dev** [nishide-dev/unturtle-legacy#91]
## 背景

`dev/benchmark_loss.py` は **loss kernel 単体** のスループットを測定するが、
実際の学習パイプライン全体 (forward + bidirectional attention + loss + backward) の
throughput は未計測。

以下が「速い」かどうかが定量的に不明:
1. A2D bidirectional Flash Attention vs SDPA fallback
2. packed sequence あり/なし
3. `DiffusionTrainer` 全体の tokens/sec (vs 素の `Trainer`)

## 目的

`dev/benchmark_training.py` を新設し、学習パイプライン全体の tokens/sec を計測する。

## Acceptance Criteria

- [ ] `dev/benchmark_training.py` を新規作成
- [ ] 対象: A2D LLaMA (tiny モデル、実 checkpoint 不要) + `DiffusionTrainer`
- [ ] 計測項目:
  - `tokens/sec` (forward + loss + backward / no optimizer step)
  - peak GPU memory (MB)
- [ ] 比較軸:
  - attention backend: Flash varlen / SDPA (packed mask) / SDPA (causal fallback)
  - packed sequence: あり / なし
  - vocab size: 32K / 128K
- [ ] 結果を Markdown 表として stdout 出力（`benchmark_loss.py` と同形式）
- [ ] 結果を CLAUDE.md の新セクション「学習スループットベンチマーク」に追記

## 実装方針

```python
# tiny A2D LLaMA config (randinit, no HF Hub)
cfg = A2DLlamaConfig(vocab_size=V, hidden_size=256, num_layers=2, ...)
model = A2DLlamaLMHeadModel(cfg).cuda().bfloat16()

# warmup → timed loop → tokens/sec
for _ in range(warmup):
    loss = model(**batch).loss
    loss.backward()
torch.cuda.synchronize()
# ... timed iters
```

## 関連

- `dev/benchmark_loss.py` (既存 loss kernel ベンチマーク)
- `tests/models/test_a2d.py` (A2D モデル構築パターンの参考)
- Issue #89 (ベンチマーク再実行)
